server/update-user.js

```diff
@@ -79,21 +79,23 @@ export default function updateUserFactory(analyticsClient) {
     const context = { active: false, ip: 0 };
     let ret = false;
 
+    // First check to see if the user is in a syncronized segments
+    // if it is not, there is no reason to keep going, just exit now...
+    const segment_ids = _.map(segments, "id");
+    if (
+      !ignoreFilters &&
+      synchronized_segments.length > 0 && // Should we move to "Send no one by default ?"
```

Comment:
In order to cleanup legacy one off cases, and to comply with the sane defaults we defined for connectors and the issue  https://github.com/hull/hull-connectors-issues/issues/49, i recommend we use this opportunity to remove this statement. 

The one thing to check is which connector instances rely on empty segments to notify the account of the change and request them to specify segments.

Reply:
I can help with doing the check before rolling out this fix.

Reply:
Cool, yea, I added a comment there too.  Seemed like we might want to update this

Reply:
I think Michal and Romain talked about this, but sounds like we're going to roll these things out in 2 different releases.  For the record, I agree with this approach.  Sometimes features are so big and tied at deployment time that it's annoying to push both out separately, but it doesn't seem so in this case. So going to roll filtering out first, then do the follow on fix for not sending anything when no segments have been defined